DemoApp/Sources/Components/Stats/ParticipantStatsView.swift

```diff
@@ -11,91 +11,114 @@ struct ParticipantStatsView: View {
     @StateObject var viewModel: ParticipantStatsViewModel
     @Injected(\.colors) var colors
     @Injected(\.fonts) var fonts
-        
-    var gridItem: GridItem {
-        GridItem.init(
-            .adaptive(minimum: itemSize + 2 * spacing),
-            spacing: spacing,
-            alignment: .topLeading
-        )
-    }
-    
-    init(call: Call, participant: CallParticipant) {
+
+    private var presentationBinding: Binding<Bool>
+    private var availableFrame: CGRect
+    private var spacing: CGFloat = 8
+
+    init(
+        call: Call,
+        participant: CallParticipant,
+        presentationBinding: Binding<Bool>,
+        availableFrame: CGRect
+    ) {
         _viewModel = StateObject(
             wrappedValue: ParticipantStatsViewModel(
                 call: call,
                 participant: participant
             )
         )
+        self.presentationBinding = presentationBinding
+        self.availableFrame = availableFrame
+        log.debug("[Stats]availableSize:\(availableFrame.size)")
     }
-    
+
     var body: some View {
-        LazyVGrid(columns: [gridItem, gridItem], spacing: spacing) {
-            ForEach(viewModel.statsEntries) { entry in
-                VStack(spacing: spacing) {
-                    Text(entry.title)
-                        .font(fonts.caption1)
-                        .lineLimit(1)
-                    Text(entry.value)
-                        .font(valueFont)
+        ScrollView {
+            LazyVGrid(columns: [.init(.adaptive(minimum: itemSize.width))], spacing: spacing) {
+                if viewModel.statsEntries.isEmpty {
+                    tileView {
+                        VStack {
+                            Text("Fetching stats...")
```

Comment:
This would stay like this even on an error.

Reply:
There is no logic currently to handle errors when loading stats tbh and i didn't spent time on this one. Maybe an improvement for later (especially given that this one is only available on the demo app)

Reply:
Ok it seems i had many safeties for the same issue. The GeometryReader when presenting this screen starts with zero frame. I have encapsulated this path with a loading message